packages/data-provider/src/services/submittedData/viewMode.ts

```diff
@@ -0,0 +1,220 @@
+import type { Schema } from '@overture-stack/lectern-client';
+
+import type { BaseDependencies } from '../../config/config.js';
+import submittedRepository from '../../repository/submittedRepository.js';
+import { generateHierarchy, type TreeNode } from '../../utils/dictionarySchemaRelations.js';
+import { pluralizeSchemaName } from '../../utils/submissionUtils.js';
+import { groupByEntityName } from '../../utils/submittedDataUtils.js';
+import { type DataRecordNested, ORDER_TYPE, type SubmittedDataResponse } from '../../utils/types.js';
+
+const viewMode = (dependencies: BaseDependencies) => {
+	const LOG_MODULE = 'VIEW_MODE_SERVICE';
+	const submittedDataRepo = submittedRepository(dependencies);
+	const recordHierarchy = dependencies.features?.recordHierarchy;
+	const { logger } = dependencies;
+
+	const convertRecordsToCompoundDocuments = async ({
+		dictionary,
+		records,
+		defaultCentricEntity,
+	}: {
+		dictionary: Schema[];
+		records: SubmittedDataResponse[];
+		defaultCentricEntity?: string;
+	}) => {
+		// get dictionary hierarchy structure
+		const hierarchyStructureDesc = generateHierarchy(dictionary, ORDER_TYPE.Values.desc);
+		const hierarchyStructureAsc = generateHierarchy(dictionary, ORDER_TYPE.Values.asc);
+
+		return await Promise.all(
+			records.map(async (record) => {
+				try {
+					const childNodes = await traverseChildNodes({
+						data: record.data,
+						entityName: record.entityName,
+						organization: record.organization,
+						schemaCentric: defaultCentricEntity || record.entityName,
+						treeNode: hierarchyStructureDesc,
+					});
+
+					const parentNodes = await traverseParentNodes({
+						data: record.data,
+						entityName: record.entityName,
+						organization: record.organization,
+						schemaCentric: defaultCentricEntity || record.entityName,
+						treeNode: hierarchyStructureAsc,
+					});
+
+					record.data = { ...record.data, ...childNodes, ...parentNodes };
+				} catch (error) {
+					logger.error(`Error converting record ${record.systemId} into compound document`, error);
+				}
+				return record;
```

Comment:
I think we should be returning some error information that the conversion failed... We don't want the client to think there was nothing to join with when that operation actually failed.

Reply:
Changed to not return any records instead it throws a 500 Http error with a message `An error occurred while converting records into compound view`.  This way user will know data or request is not valid.
Logs will describe the error details.